### PR TITLE
Use async InstanceTracker methods in SchedulerActor

### PR DESCRIPTION
**Pull request by janisz**
Releated to #4802, #3031, #4693

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/mesosphere/marathon/pull/4828" target="_blank">ReviewNinja

**Comment by jdef** [comment]
ok to test

**Comment by jdef** [comment]
thanks for the PR! maybe some conflicts with https://phabricator.mesosphere.com/D317

want to rebase?

**Comment by janisz** [comment]
Rebased

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 1]
```diff
@@ -452,9 +452,9 @@ class SchedulerActions(
     * Make sure the runSpec is running the correct number of instances
     */
   // FIXME: extract computation into a function that can be easily tested
-  def scale(runSpec: RunSpec): Unit = {
+  def scale(runSpec: RunSpec): Future[Unit] = async {
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: The scapegoat analysis tool does not accept accept `asyn/await` (problem in scapegoat) Please add a ` @SuppressWarnings` annotation. (Check with `sbt scapegoat`)

Comment (janisz) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Done

**Pull request merged**